webpack.config.js

```diff
@@ -16,14 +16,7 @@ const config = {
 		libraryTarget: "umd",
 		umdNamedDefine: true,
 	},
-	externals: (context, request, callback) => {
-		// every 'd3-*' import, will be externally required as 'd3'
-		if (/^d3-/.test(request)) {
-			return callback(null, "d3");
-		}
-
-		callback();
-	},
+	externals: /^d3-/,
```

Comment:
This is okay for module based import, but it won't be working on vanilla for those who using directly loading `d3.js`.

Reply:
Hmm, yea I'm not sure the answer there. Maybe also build a vanilla file that already includes the needed modules?

Reply:
The output of this, will be as below, requiring all d3's module as `d3-MODULE-NAME` from the root.
```js
// required external module in this PR conf
var a = typeof exports === 'object' ? factory(require("d3-time-format"), require("d3-selection"), ... ) : factory(root["d3-time-format"], root["d3-selection"], ... );

// the current
var a = typeof exports === 'object' ? factory(require("d3")) : factory(root["d3"]);
```
Even though loading each script separately,
```html
...
<script src="https://d3js.org/d3-scale.v2.min.js"></script>
```
will not work, because also in this case `d3 global is exported`.
- https://github.com/d3/d3-scale#installing

According the [webpack's conf doc](https://webpack.js.org/configuration/externals/#object), it can specify each module's export name according the module system.

```js
externals : {
    d3-selection : {
      commonjs: "d3-selection",
      amd: "d3-selection",
      root: "d3"
    },
    ...
},
```

But it seems little bit annoying declaring all d3 modules like this way. I'll be trying to figure out if there's alternative(or better) way handling this.